apis/v1alpha2/validation/gateway.go

```diff
@@ -55,8 +82,12 @@ func validateGatewayListeners(listeners []gatewayv1a2.Listener, path *field.Path
 func validateListenerHostname(listeners []gatewayv1a2.Listener, path *field.Path) field.ErrorList {
 	var errs field.ErrorList
 	for i, h := range listeners {
+		tcpUdpProtocols := []gatewayv1a2.ProtocolType{gatewayv1a2.TCPProtocolType,gatewayv1a2.UDPProtocolType}
 		// When unspecified, “”, or *, all hostnames are matched.
 		if h.Hostname == nil || (*h.Hostname == "" || *h.Hostname == "*") {
```

Comment:
Not introduced by your PR, but would be good to fix while in the vicinity.
```suggestion
		if h.Hostname == nil {
```

Reply:
can hostname not have a wildcard character ? Will the empty string be translated to nil ?

Reply:
Have not done this change in the latest patch as I am unsure whether this should be removed as comments have already been added by someone for empty and * hostnames , tests are also present for same

Reply:
This was changed in response to sig-network review since "*" was identical to empty/unspecified, it did not make sense to allow the value. Here's the updated regex validation: https://github.com/kubernetes-sigs/gateway-api/blob/5ac109cf2d1c98006e2672e4484e352ec320a412/apis/v1alpha2/shared_types.go#L263.

Reply:
Fixed